### DISCUSSION TITLE
Failed to set up: could not authenticate

### DISCUSSION THREAD

**Discussion by InvertJuha** [alandtse/tesla#322] [Q&A] [answered]
Hello,

After upgrading to latest version I got this error message:
![image](https://user-images.githubusercontent.com/40245372/199402891-dd768d94-604e-4a7d-a83d-55fd63331dde.png)

Tried to add several times, rebootet HA without luck. Double checked email address and refresh token.

Any ideas?

**Comment by cydrezz** [comment, marked as answer]
same here

**Reply by cydrezz** [reply]
did you try the second token ?
SSO Refresh Token.
Worked for me

**Reply by InvertJuha** [reply]
Well, I tried with mobile phone and it just worked well. Not really know if I had wrong token (should tried right one) but one possibility that some extra character ended up in the string while sending it to gmail and copying from there.

So now it is working. Thanks for comment and peer support. 👍
